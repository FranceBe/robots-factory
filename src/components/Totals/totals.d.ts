export type TotalsProps = {
  type?: 'primary' | 'secondary'
  label: string | JSX.Element
  total: number
}
