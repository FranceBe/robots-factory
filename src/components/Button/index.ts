import ButtonDefault, { Button } from 'components/Button/Button.component'

export { Button }

export default ButtonDefault
