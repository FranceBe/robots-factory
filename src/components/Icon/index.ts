import { FAIconByIconProps, Icon } from 'components/Icon/Icon.component'

export { FAIconByIconProps, Icon }
