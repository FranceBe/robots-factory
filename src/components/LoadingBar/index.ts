import { LoadingBar } from 'components/LoadingBar/LoadingBar.component'

export { LoadingBar }
