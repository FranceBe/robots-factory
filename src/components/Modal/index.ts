import ModalDefault, { Modal } from 'components/Modal/Modal.component'

export { Modal }

export default ModalDefault
