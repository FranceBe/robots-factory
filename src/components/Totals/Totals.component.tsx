import { TotalsContainer } from 'components/Totals/totals.style'
import { TotalsProps } from 'components/Totals/types/totals'
import React from 'react'

export const Totals: React.FC<TotalsProps> = ({ type = 'primary', total, label }) => {
  return (
    <TotalsContainer type={type}>
      <div>{label}</div>
      <div>{total}</div>
    </TotalsContainer>
  )
}

export default Totals
