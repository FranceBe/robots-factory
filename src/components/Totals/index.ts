import TotalsDefault, { Totals } from 'components/Totals/Totals.component'

export { Totals }

export default TotalsDefault
