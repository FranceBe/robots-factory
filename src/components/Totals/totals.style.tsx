import { TotalsProps } from 'components/Totals/types/totals'
import React from 'react'
import styled from 'styled-components'
import { fontSizes, palette, spaces } from 'styles/variables'

export const TotalsContainer = styled((props) => <div {...props} />)<Partial<TotalsProps>>`
  font-size: ${fontSizes.medium};
  color: ${palette.white};
  background-color: ${(props) =>
    props.type === 'primary' ? palette.pink_primary : palette.blue_primary};
  padding: ${spaces.small} ${spaces.regular};
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-radius: ${spaces.x_small};
`
