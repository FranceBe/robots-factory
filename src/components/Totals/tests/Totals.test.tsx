import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { render, screen } from '@testing-library/react'
import { Totals } from 'components/Totals'
import { TotalsProps } from 'components/Totals/totals'
import React from 'react'
import { palette } from 'styles/variables'
import { ThemeType } from 'utils/common.enum'

describe('Totals component', () => {
  const initTest = (props?: Partial<TotalsProps>) =>
    render(<Totals {...props} label={'Label'} total={10} />)

  describe('Common behaviour', () => {
    it('should match snapshot', () => {
      const { container } = initTest()

      expect(container.firstChild).toMatchSnapshot()
    })
    it('should have primary totals style by default', () => {
      const { container } = initTest()

      expect(container.firstChild).toHaveStyleRule('background-color', palette.pink_primary)
    })
    it('should display the label and the total as text', () => {
      initTest()

      expect(screen.getByText('Label')).toBeInTheDocument()
      expect(screen.getByText('10')).toBeInTheDocument()
    })
  })

  describe('Given type is "primary"', () => {
    it('should have primary background color', () => {
      const { container } = initTest({ type: ThemeType.primary })

      expect(container.firstChild).toHaveStyleRule('background-color', palette.pink_primary)
    })
  })

  describe('Given type is "secondary"', () => {
    it('should have secondary background color', () => {
      const { container } = initTest({ type: ThemeType.secondary })

      expect(container.firstChild).toHaveStyleRule('background-color', palette.blue_primary)
    })
  })
})
