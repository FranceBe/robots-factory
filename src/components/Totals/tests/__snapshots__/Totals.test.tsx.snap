// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Totals component Common behaviour should match snapshot 1`] = `
.c0 {
  font-size: 1.125rem;
  color: #FFF;
  background-color: #DB93B0;
  padding: 8px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
}

<div
  class="c0"
  type="primary"
>
  <div>
    Label
  </div>
  <div>
    10
  </div>
</div>
`;
